x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlRestValidationTestCase.java

```diff
@@ -0,0 +1,170 @@
+        "inexistent*," + indexName };
+    private static final String[] existentIndexWithoutWildcard = new String[] { indexName + ",inexistent", "inexistent," + indexName };
+    protected static final String[] existentAliasWithWildcard = new String[] {
+        aliasName + ",inexistent*",
+        aliasName + "*,inexistent*",
+        "inexistent*," + aliasName };
+    private static final String[] existentAliasWithoutWildcard = new String[] { aliasName + ",inexistent", "inexistent," + aliasName };
+    private static final String[] inexistentIndexNameWithWildcard = new String[] { "inexistent*", "inexistent1*,inexistent2*" };
+    private static final String[] inexistentIndexNameWithoutWildcard = new String[] { "inexistent", "inexistent1,inexistent2" };
+    private static final String createAlias = "{\"actions\":[{\"add\":{\"index\":\"" + indexName + "\",\"alias\":\"" + aliasName + "\"}}]}";
+    private static final String removeAlias = "{\"actions\":[{\"remove\":{\"index\":\""
+        + indexName
+        + "\",\"alias\":\""
+        + aliasName
+        + "\"}}]}";
+
+    @Before
+    @After
+    public void assertRequestBreakerEmpty() throws Exception {
+        EsqlSpecTestCase.assertRequestBreakerEmpty();
+    }
+
+    @Before
+    public void prepareIndices() throws IOException {
+        if (provisioningClient().performRequest(new Request("HEAD", "/" + indexName)).getStatusLine().getStatusCode() == 404) {
+            var request = new Request("PUT", "/" + indexName);
+            request.setJsonEntity("{\"mappings\": {\"properties\": {\"foo\":{\"type\":\"keyword\"}}}}");
+            provisioningClient().performRequest(request);
+        }
+        assertOK(provisioningAdminClient().performRequest(new Request("POST", "/" + indexName + "/_refresh")));
+    }
+
+    @After
+    public void wipeTestData() throws IOException {
+        try {
+            var response = provisioningAdminClient().performRequest(new Request("DELETE", "/" + indexName));
+            assertEquals(200, response.getStatusLine().getStatusCode());
+        } catch (ResponseException re) {
+            assertEquals(404, re.getResponse().getStatusLine().getStatusCode());
+        }
+    }
+
+    private String getInexistentIndexErrorMessage() {
+        return "\"reason\" : \"Found 1 problem\\nline 1:1: Unknown index ";
+    }
+
+    public void testInexistentIndexNameWithWildcard() throws IOException {
+        assertErrorMessages(inexistentIndexNameWithWildcard, getInexistentIndexErrorMessage(), 400);
+    }
+
+    public void testInexistentIndexNameWithoutWildcard() throws IOException {
+        assertErrorMessages(inexistentIndexNameWithoutWildcard, getInexistentIndexErrorMessage(), 400);
+    }
+
+    public void testExistentIndexWithoutWildcard() throws IOException {
+        for (String indexName : existentIndexWithoutWildcard) {
+            assertErrorMessage(indexName, "\"reason\" : \"no such index [inexistent]\"", 404);
```

Comment:
Should we have a 400 for `inexistentIndexNameWithoutWildcard` but a 404 if the same index wasn't found as part of a pattern from `existentIndexWithoutWildcard`?

Reply:
`from existent, inexistent` with `404` (`existentIndexWithoutWildcard`) comes from ES itself.
`from inexistent` with `400` (`inexistentIndexNameWithoutWildcard`) comes from ES|QL.

Reply:
Hm, I wonder if ESQL should just catch the errors from ES to provide consistent `ValidationException`s, and consistent error codes.

Reply:
Created issue https://github.com/elastic/elasticsearch/issues/112132